app/models/pacbio/pool.rb

```diff
@@ -22,6 +22,22 @@ class Pool < ApplicationRecord
               :insert_size, numericality: { greater_than_or_equal_to: 0, allow_nil: true }
 
     validates :used_aliquots, presence: true
+    validates :primary_aliquot, presence: true
+    validate :used_aliquots_volume, if: lambda {
+                                          Flipper.enabled?(:dpl_1072_check_library_volume_in_pools)
+                                        }
+
+    accepts_nested_attributes_for :primary_aliquot
+
+    def used_aliquots_volume
+      # If the aliquot is from a library we want to check there is enough volume
+      used_aliquots.select { |aliquot| aliquot.source_type == 'Pacbio::Library' }.each do |aliquot|
+        unless aliquot.source.available_volume_sufficient(aliquot.volume)
+          errors.add(:base, "Insufficient volume available for #{aliquot.source.tube.barcode}")
+          return false
```

Comment:
Few doubts
- Is it required to go through all, even after it finds one with insufficient volume?
- Do we need to add multiple errors, or should we add an error with all barcodes?

Reply:
Yes going point, probably makes more sense to string together the barcode into one error message.